kpet/data.py

```diff
@@ -171,6 +171,36 @@ def matches_regex_list(self, target, name):
                     return True
         return False
 
+    def matches_regex_list_validate(self, target, name):
+        """
+        Check if the pattern is valid as in matches at least a single value
+
+        Args:
+            target: The target (an instance of Target) to match.
+            name:   The name of the parameter to match.
+
+        Returns:
+            None if the pattern matches for the parameter
+            dict (with information about the invalid regex) otherwise.
+        """
+        assert isinstance(target, Target)
+        assert isinstance(name, str)
+        regex_list = getattr(self, name, None)
+        value_set = getattr(target, name)
+
+        if regex_list is None or regex_list == [] or value_set == set():
+            return None
+        # Match if there are any regex/value matches
+        for regex in regex_list:
+            for value in value_set:
+                if regex.fullmatch(value):
+                    break
+            else:
+                return {"pattern": regex, "dataset": value_set,
+                        "match": "match" if self.positive else "dont_match"}
```

Comment:
The `Pattern` class shouldn't know anything about `match` and `dont_match`. Only the `Base` class knows what those are, and we create opportunity for confusion and mismatch by duplicating that here. So, drop this field and leave figuring this out to the `Base` class.